azurerm/resource_arm_media_services_test.go

```diff
@@ -0,0 +1,252 @@
+			Type: media.Primary,
+		},
+		{
+			ID:   utils.String("id2"),
+			Type: media.Primary,
+		},
+	}
+
+	actualErr := validateStorageConfiguration(accounts)
+	expectedErr := fmt.Errorf("Error processing storage account 'id2'. Another storage account is already assigned as is_primary = 'true'")
+
+	if actualErr == nil || expectedErr.Error() != actualErr.Error() {
+		t.Fatal("validateStorageConfiguration must throw an error when more than one storage account is Primary.")
+	}
+
+	accounts = []media.StorageAccount{
+		{
+			ID:   utils.String("id1"),
+			Type: media.Primary,
+		},
+		{
+			ID:   utils.String("id2"),
+			Type: media.Secondary,
+		},
+	}
+
+	actualErr = validateStorageConfiguration(accounts)
+
+	if actualErr != nil {
+		t.Fatal("validateStorageConfiguration must allow multiple storage accounts when only one is Primary.")
+	}
+}
+
+func testCheckAzureRMMediaServicesAccountExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Media service not found: %s", resourceName)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Media Services Account: '%s'", name)
+		}
+
+		conn := testAccProvider.Meta().(*ArmClient).mediaServicesClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resp, err := conn.Get(ctx, resourceGroup, name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on mediaServicesClient: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: Media Services Account '%s' (resource group: '%s') does not exist", name, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testCheckAzureRMMediaServicesAccountDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).mediaServicesClient
+	ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_media_services" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(ctx, resourceGroup, name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("Media Services Account still exists:\n%#v", resp)
+		}
+	}
+
+	return nil
+}
+
+func checkAccAzureRMMediaServicesAccount(resourceName string, location string, storageCount string) resource.TestCheckFunc {
+	// It would be ideal to also validate which storage account was Primary, but that isn't straight forward.
+	// The key for the storage account's Primary flag is non-deterministic (ex: storage_account.1137153885.is_primary) and
+	// isn't based on the storage account name.
```

Comment:
heh indeed, that said we should be able to remove this since the `import` stage verifies the config matches the local state

Reply:
@tombuildsstuff - I disappeared for a week to work on some other stuff. Can you let me know if I'm misunderstanding the two remaining active comments?